src/test/java/com/contrastsecurity/maven/plugin/it/stub/FakeContrastAPI.java

```diff
@@ -232,6 +310,91 @@ private static void startScan(final HttpExchange exchange) {
     exchange.close();
   }
 
+  private static void scan(final HttpExchange exchange) {
+    discardRequest(exchange);
+    final String json =
+        "{\n"
+            + "  \"id\": \""
+            + SCAN_ID
+            + "\",\n"
+            + "  \"organizationId\": \""
+            + ORGANIZATION_ID
+            + "\",\n"
+            + "  \"projectId\": \""
+            + PROJECT_ID
+            + "\",\n"
+            + "  \"codeArtifactId\": \""
+            + CODE_ARTIFACT_ID
+            + "\",\n"
+            + "  \"status\": \"COMPLETED\",\n"
+            + "  \"createdTime\": \"2021-07-26T18:45:19.222+00:00\",\n"
+            + "  \"startedTime\": \"2021-07-26T18:46:00\",\n"
+            + "  \"completedTime\": \"2021-07-25T18:50:00\",\n"
+            + "  \"language\": \"JAVA\",\n"
+            + "  \"label\": \"scan.http\",\n"
+            + "  \"errorMessage\": null,\n"
+            + "  \"includeNamespaceFilters\": [],\n"
+            + "  \"excludeNamespaceFilters\": []\n"
+            + "}";
+    final byte[] body = json.getBytes(StandardCharsets.UTF_8);
+    try {
+      exchange.sendResponseHeaders(201, body.length);
+      exchange.getResponseBody().write(body);
+    } catch (final IOException e) {
+      throw new UncheckedIOException("Failed to send response", e);
+    }
+    exchange.close();
```

Comment:
I wasted a lot of time figuring out how to close the exchange correctly. In like the idea of wrapping in something AutoCloseable. I'll make that change and reduce the boilerplate.

Reply:
`HttpExchange` is `AutoCloseable`, but since we didn't make the object, and since we aren't using Java 9, we can't use it in a try-with-resources. We could if we wrap it as you're recommending. Simply closing it is not good enough. I found we also need to consume the whole request before trying to send the response, and the `close()` method doesn't necessarily read the whole request (hence the discardRequest method). 

If we like the "extract JSON to external file" recommendation from a previous thread, then I think I have a solution to reduce the boilerplate.

Reply:
Sounds good. I don’t know if this fits in with that, but we could wrap the `discardRequest` functionality into the `AutoCloseable`:

```
  private static final class RequestDiscardingHttpExchangeWrapper implements AutoCloseable {
    private final HttpExchange httpExchange;

    private CloseableHttpExchangeWrapper(final HttpExchange httpExchange) {
      this.httpExchange = httpExchange;
    }
    
    private HttpExchange httpExchange() {
      return httpExchange;
    }

    @Override
    public void close() throws Exception {
      final byte[] buffer = new byte[4096];
      try (InputStream is = httpExchange.getRequestBody()) {
        //noinspection StatementWithEmptyBody
        while (is.read(buffer) >= 0) {}
      } catch (final IOException e) {
        throw new UncheckedIOException("Failed to read request", e);
      } finally {
        httpExchange.close();
      }
    }
  }
```

Or maybe make a full delegate of `RequestDiscardingHttpExchange` and just change the `close` method. Or we could handle closing the exchange in a wrapper lambda.

Reply:
I learned that discarding the request in `close()` is too late. The request has to be fully read before we start writing the response.